src/app/components/register/register.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, OnInit } from '@angular/core';
+import { RegisterService } from '../../services/register/register.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  username: string;
+  password: string;
+  confirmPsw: string;
+  usernameError: string;
+  passwordError: string;
+  confirmPswError: string;
+  messageSuccess: string;
+  registerStatus: boolean;
+
+  constructor(private registerService: RegisterService, private router: Router) { }
+
+  ngOnInit() { }
+
+  CheckUserName(username: string): boolean {
+    if (username !== undefined && username.length > 0) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  myValidateFunction(username: string) {
+    const pattern = /^[A-Za-z0-9]\w{0,}$/;
+    if (pattern.test(username)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  CheckPassword(password: string): boolean {
+    if (password !== undefined && password.length > 7) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  CheckConfirmPassword(password: string, passwordAgain: string): boolean {
+    if (password !== undefined && passwordAgain !== undefined && password === passwordAgain) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  registerUser() {
+    if (this.CheckUserName(this.username) === false) {
+      this.usernameError = 'Username must be at least 1 character';
+      this.registerStatus = false;
+    } else if (this.myValidateFunction(this.username) === false) {
+      this.usernameError = 'Username can not contain any special character';
+      this.registerStatus = false;
+    } else {
+      this.usernameError = '';
+      this.registerStatus = true;
+    }
+    if (this.CheckPassword(this.password) === false) {
+      this.passwordError = 'Password must be at least 8 characters';
+      this.registerStatus = false;
+    } else {
+      this.passwordError = '';
+      this.registerStatus = true;
+    }
+    if (this.CheckConfirmPassword(this.password, this.confirmPsw) === false) {
+      this.confirmPswError = 'Password confirmation must be the same';
+      this.registerStatus = false;
+    } else {
+      this.confirmPswError = '';
+      this.registerStatus = true;
```

Comment:
if my username is wrong but the passwords match, the registerStatus still will be true because of this, no?
you should find a solution that doesn't ignore previous mistakes.

Reply:
You are right. I added 3 statuses instead of 1 and 1 checkStatus function and now it works in every case